### PR TITLE
GITOPSRVCE-704: Improve test coverage of 'application_event_runner.go' in backend component

### PR DESCRIPTION
**Pull request by jgwest**
#### Description:
- Vast majority of new work in this PR is in 'Generic Application Event Loop Runner tests' in `application_event_runner_test.go`
- The rest of the PR is just some misc refactoring:
    - In a couple cases, I changed `chan *eventlooptypes.EventLoopEvent` to `chan eventlooptypes.EventLoopEvent`, as it didn't make sense for these to be pointers.
    - Added context and factory parameters to ApplicationEventLoopRunner interface

### CODE REVIEW

**Comment by jgwest** [comment]
/retest
There's already a JIRA issue open for the intermittent failure

File: backend/eventloop/application_event_loop/application_event_runner_test.go [hunk 1]
```diff
@@ -1225,6 +1226,740 @@ var _ = Describe("ApplicationEventLoop Test", func() {
...
+				Expect(applId).ToNot(BeEmpty(), "Application row should exist in the DB, for the initial GitOpsDeployment")
+
+			})
+		})
+
+		When("the application event loop runner receives a GitOpsDeploymentSyncRun event for a GitOpsDeployment", func() {
+
+			It("verifies the event is processed", func() {
+
+				By("creating a generic GitOpsDeployment of type manual sy=nc")
```
Comment (Rizwana777) [inline backend/eventloop/application_event_loop/application_event_runner_test.go:1415]: typo : sy=nc -> sync

File: backend/eventloop/application_event_loop/application_event_runner_test.go [hunk 2]
```diff
@@ -1225,6 +1226,740 @@ var _ = Describe("ApplicationEventLoop Test", func() {
...
+						select {
+						case <-ctx.Done():
+							return
+						default:
+						}
+
+						// wait for Applicaton row to exist for GitopsDeployment
+						applId := findDBApplicationIDforGitOpsDeployment(ctx, gitopsDeploymentName, gitopsDeploymentNamespace, *namespace)
+						if applId == "" {
+							fmt.Println("waitin for application for gitopsdepl")
```
Comment (Rizwana777) [inline backend/eventloop/application_event_loop/application_event_runner_test.go:1690]: typo : waitin -> waiting

File: backend/eventloop/application_event_loop/application_event_runner_test.go [hunk 3]
```diff
@@ -1225,6 +1226,740 @@ var _ = Describe("ApplicationEventLoop Test", func() {
...
+						// End the goroutine on cancel
+						select {
+						case <-ctx.Done():
+							return
+						default:
+						}
+
+						applId := findDBApplicationIDforGitOpsDeployment(ctx, gitopsDeploymentName, gitopsDeploymentNamespace, *namespace)
+						if applId == "" {
+							fmt.Println("waitin for application for gitopsdepl")
```
Comment (Rizwana777) [inline backend/eventloop/application_event_loop/application_event_runner_test.go:1842]: typo : waitin -> waiting

**Comment by Rizwana777** [comment]
@jgwest Looks good to me, added comments for few typos

**Comment by jgwest** [comment]
Thanks @Rizwana777!

**Pull request merged**